client/options.go

```diff
@@ -134,7 +134,11 @@ func WithScheme(scheme string) Opt {
 }
 
 // WithTLSClientConfig applies a TLS config to the client transport.
-func WithTLSClientConfig(cacertPath, certPath, keyPath string) Opt {
+func WithTLSClientConfig(cacertPath, certPath, keyPath string, skipVerify ...bool) Opt {
+	insecureSkipVerify := false
+	if len(skipVerify) > 0 {
+		insecureSkipVerify = skipVerify[0]
+	}
```

Comment:
This should either use the `Opts` pattern that would allow to perform arbitrary modifications to the `tlsconfig.Options`, or just add a new function that accepts a whole `tlsconfig.Options`.